### DISCUSSION TITLE
API Docs differ from API Response

### DISCUSSION THREAD

**Discussion by MajesticFalcon** [netbox-community/netbox#10866] [General]
When I go to /api/docs and expand Devices, I see a parameter for "power_ports". When I run a query to /api/dcim/devices/ID, I do not receive a response including "power_ports". Am I missing something?

**Comment by candlerb** [comment]
Netbox version?

> When I go to /api/docs and expand Devices, I see a parameter for "power_ports".

Are you looking specifically at "GET /dcim/devices (dcim_devices_list)"? If so, you'll see that "power_ports" is a query parameter, not part of the response. This means that you could make an API call to `/api/dcim/devices/?power_ports=XXX` to limit the list of devices returned.

What value is meaningful for `XXX` here? The API docs only say this is a "string" value which doesn't help at all, so I'm afraid you have to go dig through the source code looking for "filtersets". The answer is [here](https://github.com/netbox-community/netbox/blob/develop/netbox/dcim/filtersets.py#L932-L935): it's a boolean flag, i.e. you can select devices which have power ports or devices which don't have power ports.

```
# curl -gsS -H "Authorization: Token $API_TOKEN" "https://netbox.example.net/api/dcim/devices/" |  jq '.results[].id' | wc -l
20
# curl -gsS -H "Authorization: Token $API_TOKEN" "https://netbox.example.net/api/dcim/devices/?power_ports=true" | jq '.results[].id' | wc -l
12
# curl -gsS -H "Authorization: Token $API_TOKEN" "https://netbox.example.net/api/dcim/devices/?power_ports=false" | jq '.results[].id' | wc -l
8
# 
```

**Reply by MajesticFalcon** [reply]
Thanks Brian!
